sdk/storage/Azure.Storage.Blobs.Cryptography/src/EncryptedBlobClient.cs

```diff
@@ -0,0 +1,323 @@
+        /// name of the account, the name of the container, and the name of
+        /// the blob.
+        /// </param>
+        /// <param name="pipeline">
+        /// The transport pipeline used to send every request.
+        /// </param>
+        internal EncryptedBlobClient(Uri blobUri, HttpPipeline pipeline)
+            : base(blobUri, pipeline)
+        {
+            //TODO make new pipeline for this one
+        }
+        #endregion ctors
+
+        /// <summary>
+        /// This method performs a transform on the content stream for uploads. It is a no-op by default.
+        /// </summary>
+        /// <param name="content">Content to transform.</param>
+        /// <param name="metadata">Content metadata to transform.</param>
+        /// <returns>Transformed content stream.</returns>
+        internal protected override (Stream, Metadata) TransformContent(Stream content, Metadata metadata)
+        {
+            metadata ??= new Dictionary<string, string>();
+
+            var (encryptionStream, encryptionData) = EncryptStream(content);
+
+            using (var stream = new MemoryStream())
+            {
+                var serializer = new DataContractJsonSerializer(typeof(EncryptionData));
+                serializer.WriteObject(stream, encryptionData);
+                stream.Position = 0;
+                metadata.Add(EncryptionConstants.ENCRYPTION_DATA_KEY, new StreamReader(stream).ReadToEnd());
+            }
+
+            return (encryptionStream, metadata);
+        }
+
+        private (Stream, EncryptionData) EncryptStream(Stream plaintext)
+        {
+            var generatedKey = CreateKey(EncryptionConstants.ENCRYPTION_KEY_SIZE_BITS);
+
+            using (AesCryptoServiceProvider aesProvider = new AesCryptoServiceProvider() { Key = generatedKey })
+            {
+
+                var encryptionData = new EncryptionData()
+                {
+                    EncryptionMode = EncryptionConstants.ENCRYPTION_MODE,
+                    ContentEncryptionIV = aesProvider.IV,
+                    EncryptionAgent = new EncryptionAgent()
+                    {
+                        EncryptionAlgorithm = ClientsideEncryptionAlgorithm.AES_CBC_256,
+                        Protocol = EncryptionConstants.ENCRYPTION_PROTOCOL_V1
+                    },
+                    KeyWrappingMetadata = new Dictionary<string, string>()
+                    {
+                        { EncryptionConstants.AGENT_METADATA_KEY, EncryptionConstants.AGENT_METADATA_VALUE }
+                    },
+                    WrappedContentKey = new WrappedKey()
+                    {
+                        Algorithm = null, // TODO
+                        EncryptedKey = this.KeyWrapper.WrapKey(KeyWrapAlgorithm, generatedKey).ToArray(),
+                        KeyId = this.KeyWrapper.KeyId
+                    }
+                };
+
+                return (
+                    new RollingBufferStream(
```

Comment:
Can we pull this out into a separate variable?  The implicit tuple syntax gets a little lost like this.